gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/AbstractSpecExecutor.java

```diff
@@ -182,7 +183,7 @@ public int hashCode() {
 
   abstract protected void shutDown() throws Exception;
 
-  abstract public Future<? extends SpecProducer> getProducer();
+  abstract public Future<? extends SpecProducer<Spec>> getProducer();
```

Comment:
Will this have any impact on the serialization/deserialization of Dags in the DagManager mode?

Reply:
I don't think so. We are actually not serializing/deserializing SpecProducer. We serialize/deserialize SpecExecutor and then find the producer from a mapping. After finding a producer we serialize/deserialize the onSpecAdd response Future.